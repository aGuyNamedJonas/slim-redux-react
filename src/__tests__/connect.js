import React from 'react';
import { connect, Provider } from '../';
import { shallow } from 'enzyme';
import { changeTrigger as slimReduxChangeTrigger, createSlimReduxStore } from 'slim-redux';

const store      = createSlimReduxStore({ one: 'one', two: 'two', three: { four: 'four' } }),
      resetStore = slimReduxChangeTrigger('RESET_STORE', state => ({ one: 'one', two: 'two', three: { four: 'four' } }));

const BasicComponent = (props) => (
  <div>Basic Component</div>
);

beforeEach(() => resetStore());

describe('connect() (default behavior)', () => {
  test('returns a react component', () => {
    const ConnectedBasicComponent = connect(BasicComponent), 
          wrapper = shallow(
            <Provider store={store}>
              <ConnectedBasicComponent/>
            </Provider>
          );

    expect(wrapper).toMatchSnapshot();
  });

  test('makes subscription() and calculation() values available to connected component through props', () => {
    expect(true).toBe(false);
  });

  test('re-render when subscription changes', () => {
    expect(true).toBe(false);
  });

  test('does not re-render when non-subscribed to part of the state changes', () => {
    expect(true).toBe(false);
  });

  test('re-renders when calculation changes', () => {
    expect(true).toBe(false);
  });

  test('does not re-render when a part of the state changes that the calculation is not subscribed to', () => {
    expect(true).toBe(false);
  });

  test('makes change triggers available to component through props', () => {
    expect(true).toBe(false);
  });

  test('makes async change triggers available to component through props', () => {
    expect(true).toBe(false);
  });

  test('subscriptions, calculations, change triggers and async change triggers use the store instance provided through the <Provider/> component', () => {
    expect(true).toBe(false);
  });
});
