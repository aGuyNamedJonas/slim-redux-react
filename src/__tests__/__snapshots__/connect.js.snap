// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`connect() (default behavior) returns a react component 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <SlimReduxConnectorBasicComponent />,
  "nodes": Array [
    <SlimReduxConnectorBasicComponent />,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <Provider
        store={
                Object {
                        "dispatch": [Function],
                        "getState": [Function],
                        "registerChangeTrigger": [Function],
                        "replaceReducer": [Function],
                        "slimReduxChangeTriggers": Object {
                          "RESET_STORE": Object {
                            "focusSubString": null,
                            "getFocusState": null,
                            "reducer": [Function],
                            "setFocusState": null,
                          },
                        },
                        "slimReduxOptions": Object {
                          "disableActionDispatch": false,
                          "disableGlobalStore": false,
                        },
                        "subscribe": [Function],
                        Symbol(observable): [Function],
                      }
        }
>
        <SlimReduxConnectorBasicComponent />
</Provider>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Provider {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "children": <SlimReduxConnectorBasicComponent />,
          "store": Object {
            "dispatch": [Function],
            "getState": [Function],
            "registerChangeTrigger": [Function],
            "replaceReducer": [Function],
            "slimReduxChangeTriggers": Object {
              "RESET_STORE": Object {
                "focusSubString": null,
                "getFocusState": null,
                "reducer": [Function],
                "setFocusState": null,
              },
            },
            "slimReduxOptions": Object {
              "disableActionDispatch": false,
              "disableGlobalStore": false,
            },
            "subscribe": [Function],
            Symbol(observable): [Function],
          },
        },
        "refs": Object {},
        "state": null,
        "store": Object {
          "dispatch": [Function],
          "getState": [Function],
          "registerChangeTrigger": [Function],
          "replaceReducer": [Function],
          "slimReduxChangeTriggers": Object {
            "RESET_STORE": Object {
              "focusSubString": null,
              "getFocusState": null,
              "reducer": [Function],
              "setFocusState": null,
            },
          },
          "slimReduxOptions": Object {
            "disableActionDispatch": false,
            "disableGlobalStore": false,
          },
          "subscribe": [Function],
          Symbol(observable): [Function],
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <SlimReduxConnectorBasicComponent />,
        "_debugID": 2,
        "_renderedOutput": <SlimReduxConnectorBasicComponent />,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Provider
    store={
        Object {
            "dispatch": [Function],
            "getState": [Function],
            "registerChangeTrigger": [Function],
            "replaceReducer": [Function],
            "slimReduxChangeTriggers": Object {
              "RESET_STORE": Object {
                "focusSubString": null,
                "getFocusState": null,
                "reducer": [Function],
                "setFocusState": null,
              },
            },
            "slimReduxOptions": Object {
              "disableActionDispatch": false,
              "disableGlobalStore": false,
            },
            "subscribe": [Function],
            Symbol(observable): [Function],
          }
    }
>
    <SlimReduxConnectorBasicComponent />
</Provider>,
}
`;
